### DISCUSSION TITLE
why isn't the correct WiFi driver included in standard kernel?

### DISCUSSION THREAD

**Discussion by ghost** [linux-surface/linux-surface#474] [Q&A] [answered]
why isn't the correct WiFi driver included in standard kernel?
why is it necessary to cp it manually upon install?

**Comment by qzed** [comment]
I assume you're referring to the ath10k firmware? The driver is upstream (i.e. included in the kernel). Some people have tried (multiple times) to get that firmware upstream, but it got rejected. AFAIK we don't know the full reason for this but it might have to do with licensing issues or incompatibility for different devices.

So for now you unfortunately either have to use the [workaround](https://github.com/linux-surface/linux-surface/wiki/Surface-Laptop-3#atheros-qca6174-wifi-firmware) or copy it over manually.

**Reply by ghost** [reply]
Yes, that's what I was referring to! The firmware, not the driver. Thank you for the explanation. Very weird indeed. Do you think it may ever get into the upstream?

**Reply by qzed** [reply]
I'm a bit doubtful at this point, as the board file has now been removed from the official website, which will probably make this only harder.

**Comment by ghost** [comment, marked as answer]
It's actually been added to the kernel now, right?

**Reply by ghost** [reply]
Should we mark this discussion as answered then?

**Reply by qzed** [reply]
Yes, thanks to Hans de Goede the proper firmware is now in the official upstream repo.
